component/src/main/java/org/wso2/extension/siddhi/map/avro/sourcemapper/AvroSourceMapper.java

```diff
@@ -354,6 +445,95 @@ protected void mapAndProcess(Object eventObject, InputEventHandler inputEventHan
+     * Creation of empty byte array and length > 0 check for each event conversion is costly
+     */
+    private Event[] convertToSingleEventForCustomMapping(String avroMessage) {
+        Configuration conf = Configuration.defaultConfiguration();
+        ReadContext readContext = JsonPath.parse(avroMessage);
+        List<Event> eventList = new ArrayList<>();
+        Event event = new Event(streamAttributesSize);
+        Object[] data = event.getData();
+        Object childObject = readContext.read(DEFAULT_JSON_PATH);
+        readContext = JsonPath.using(conf).parse(childObject);
+
+        for (MappingPositionData mappingPositionData : this.mappingPositions) {
+            int position = mappingPositionData.getPosition();
+            Object mappedValue;
+            try {
+                mappedValue = readContext.read(mappingPositionData.getMapping());
+                if (mappedValue == null) {
+                    data[position] = null;
+                } else {
+                    if (streamAttributes.get(position).getType().equals(Attribute.Type.OBJECT)) {
+                        JsonParser parser = null;
+                        JsonNode jsonObjectNode;
+                        try {
+                            parser = jsonFactory.createParser(mappedValue.toString());
+                            switch (parser.nextToken()) {
+                                case START_OBJECT:
+                                case START_ARRAY:
+                                    jsonObjectNode = objectMapper.readTree(avroMessage).
+                                            findValue(mappingPositionData.getMapping());
+                                    data[position] = gson.fromJson(jsonObjectNode.toString(), Object.class);
+                                    break;
+                                case VALUE_STRING:
+                                    data[position] = parser.getValueAsString();
+                                    break;
+                                case VALUE_NUMBER_INT:
+                                    data[position] = parser.getValueAsInt();
+                                    break;
+                                case VALUE_NUMBER_FLOAT:
+                                    data[position] = attributeConverter.getPropertyValue(parser.getValueAsString(),
+                                            Attribute.Type.FLOAT);
+                                    break;
+                                case VALUE_TRUE:
+                                case VALUE_FALSE:
+                                    data[position] = parser.getValueAsBoolean();
+                                    break;
+                                default:
+                                    data[position] = null;
+                                    log.info(parser.nextToken() + " is not a acceptable data format. Hence data " +
+                                             "value is set to null");
```

Comment:
Any reason for logging it as info? Again, provide bit more context details for the error message.

Reply:
It is only needed to notify the user, that the data value is set to null. That is why it is logged as a info. 
Changes to log message is done in commit 8242c37248b009ed5114d985b9faeedc08482da2

Reply:
No, my question is about the logging level. Do you think above should be an "info" level logging?

Reply:
I have changed the log level from info to warn after referring to some logging options. Commit d82f0a3aab80353b164eec5d86cb7244e98cf916